### PR TITLE
[DRAFT] front end structure initial draft - feedback encouraged

### PR DESCRIPTION
**Pull request by alexbowen**
this is a first draft of clarification of what the front end structure should be and will drive the implementation/refactoring in upcoming JIRA tickets. this is intended for discussion/ideas/thoughts

my main drivers for this are (and this is what i encourage discussion/comments on)
- consistency
- not over engineering
- finding/using functionality is intuitive
- reusability

### CODE REVIEW

File: app/components/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+JS and SCSS files are optional
+```
+/app
+  /components
+    /my_component
+      my_component.js
+      my_component.test.js
+      my_component.scss
+      my_component.html.haml
+    my_component.rb
```
Comment (david-mears-dfe) [inline app/components/README.md:11]: Do we want the .rb in the directory, too? (Saving vertical visual space)

Comment (alexbowen) [reply app/components/README.md:11]: my thinking was thats in an entry point to using the component

Comment (alexbowen) [reply app/components/README.md:11]: in node.js you can have an index file that mean you only need to reference the folder. if can do this consistently with rails too i would favour what you suggest here @david-mears-dfe

Comment (david-mears-dfe) [reply app/components/README.md:11]: "thats in an entry point to using the component"

I'm not sure what you mean?

I'm not too picky about this.

**Review by cpjmcquillan — APPROVED** [review]
Looks sensible to me 👍

**Review by ddippolito — APPROVED** [review]
Happy with this

**Pull request merged**